components/push/src/subscriber.rs

```diff
@@ -161,6 +163,33 @@ impl PushManager {
             .map_err(|e| ErrorKind::TranscodingError(format!("{:?}", e)).into())
     }
 
+    /// Fetch new endpoints for a list of channels.
+    pub fn regenerate_endpoints(&mut self) -> error::Result<Vec<DispatchInfo>> {
+        if self.conn.uaid.is_none() {
+            return Err(ErrorKind::GeneralError("No subscriptions defined yet.".into()).into());
+        }
+        let uaid = self.conn.uaid.clone().unwrap();
```

Comment:
Transform Some to Result.

Reply:
You want `Option<T>::ok_or_else`:
https://github.com/mozilla/application-services/blob/2df37a4c8c9d3b9c9159b0f80542303088027618/components/fxa-client/src/scoped_keys.rs#L20

Reply:
Actually, I think we don't need this, since we're checking the `conn.uaid.is_none()` right above. So the `unwrap` should be quite safe.

Reply:
Yes you should replace the whole block by `ok_or_else`